frontend/src/lib/utils.ts

```diff
@@ -203,3 +207,49 @@ export function labelVisibilityProtoValueToEnum(
       return LabelVisibilityOptions.Visible
   }
 }
+
+export function isAppInLoadingState(): boolean {
+  if (document.querySelectorAll('[data-testid="stStatusWidget"]').length > 0) {
+    return true
+  }
+  const appViewContainerQS = document.querySelectorAll(
+    '[data-testid="stAppViewContainer"]'
+  )
+  if (appViewContainerQS.length > 0) {
+    const stAlertQS = appViewContainerQS[0].getElementsByClassName("stAlert")
+    if (stAlertQS.length > 0) {
+      const stAlert = stAlertQS[0]
+      const tagsQS = stAlert.querySelectorAll(
+        '[data-testid="stMarkdownContainer"]'
+      )
+      if (tagsQS.length > 0) {
+        return tagsQS[0].innerHTML === `<p>${PLEASE_WAIT_MESSAGE}</p>`
+      }
+    }
+  }
+  return false
+}
+
```

Comment:
Unit testing of this is something we need to discuss, I don't see a clear, elegant solution to unit test this, my concerns are:
* we are operating on `document.URL` mostly, which is readonly property of `document`
* I've seen some libraries which mock the DOM, however should we introduce new dependency? We've worked before to limit our dependencies is it worth to introduce a new library for a few functions?
* I think it's not elegant to mock the dom in unit tests
* another problem is: we are relying on callback here so there's a lots of things to mock
I don't know what are we going to decide, however if we really need more tests I think we need to think about E2E tests 🤔 

What do you think @sfc-gh-mnowotka ?

Reply:
Yes, you brought up a very good point! One solution for testing these functions would be to have a rule that `util` functions do not have any side effects nor rely on any global states. Instead they are just `pure` functions. This means that only rely on their input params and they only communicate with the outside world by the return value. So in the context of this PR, this would mean changing the function signature from:

```
removeURLParam(paramKey: string): void
```

to:

```
removeURLParam(url: string, paramKey: string): string
```

if you apply the same to all the functions you defined in util then you'll be able to test them. Then pass the specific url inside the callback.

Reply:
A broader problem is how to check if printing itself actually works as expected and for this I have too little knowledge of our testing tools.

Reply:
> A broader problem is how to check if printing itself actually works as expected and for this I have too little knowledge of our testing tools.

Is it worth it to test it? Maybe QA is enough, most of the code presented in the PR is tested, after pure functions were introduced. @jrieke do you think we are good with this change with QA testing?

Reply:
Yeah I think there's probably little we can test automatically. I'm fine with just doing QA for this. If issues do come up, we can still see if we can improve later on.